packages/js/product-editor/src/components/header/publish-button/publish-button.tsx

```diff
@@ -6,39 +6,35 @@ import { Button } from '@wordpress/components';
 import { createElement } from '@wordpress/element';
 import { getNewPath, navigateTo } from '@woocommerce/navigation';
 import { MouseEvent } from 'react';
-import { Product, ProductStatus } from '@woocommerce/data';
+import { Product } from '@woocommerce/data';
 import { useDispatch } from '@wordpress/data';
-import { useEntityProp } from '@wordpress/core-data';
 
 /**
  * Internal dependencies
  */
 import { getProductErrorMessage } from '../../../utils/get-product-error-message';
 import { recordProductEvent } from '../../../utils/record-product-event';
 import { usePublish } from '../hooks/use-publish';
+import { PublishButtonProps } from './types';
 
-export function PublishButton(
-	props: Omit< Button.ButtonProps, 'aria-disabled' | 'variant' | 'children' >
-) {
-	const [ productStatus ] = useEntityProp< ProductStatus >(
-		'postType',
-		'product',
-		'status'
-	);
-
-	const isCreating = productStatus === 'auto-draft';
-
+export function PublishButton( {
+	productStatus,
+	...props
+}: PublishButtonProps ) {
 	const { createSuccessNotice, createErrorNotice } =
 		useDispatch( 'core/notices' );
 
 	const publishButtonProps = usePublish( {
+		productStatus,
 		...props,
 		onPublishSuccess( savedProduct: Product ) {
 			recordProductEvent( 'product_update', savedProduct );
 
-			const noticeContent = isCreating
-				? __( 'Product successfully created.', 'woocommerce' )
-				: __( 'Product published.', 'woocommerce' );
+			const isPublished = productStatus === 'publish';
+
+			const noticeContent = isPublished
+				? __( 'Product published.', 'woocommerce' )
```

Comment:
This feels a bit weird to me and may have been a pre-existing issue or intentional (cc @jarekmorawski).

Do we want to show the notice "Product published" every time after save?  I would think updating an already published product would say "Product updated" or something similar.

Reply:
Yes. The first notice should say `Product added` (more in line with the initial CTA), and the second `Product updated`.

Reply:
Fixed -> https://github.com/woocommerce/woocommerce/pull/38507/commits/159c598d5958e12c8f181301e1aedbf832735756
Also considering -> https://github.com/woocommerce/woocommerce/issues/38202#issuecomment-1553481225